tests/test_spark_ai.py

```diff
@@ -334,18 +360,146 @@ def test_spark_connect_pivot(self):
                     ("C", "English", 90),
                     ("C", "Science", 100),
                 ],
-                ["Student", "Subject", "Marks"]
+                ["Student", "Subject", "Marks"],
             )
             result = df.ai.transform("pivot using Subject for Marks")
 
-            expected_lst = [Row(Student='B', English=75, Maths=80, Science=None),
-                            Row(Student='C', English=90, Maths=None, Science=100),
-                            Row(Student='A', English=45, Maths=50, Science=None)]
+            expected_lst = [
+                Row(Student="B", English=75, Maths=80, Science=None),
+                Row(Student="C", English=90, Maths=None, Science=100),
+                Row(Student="A", English=45, Maths=50, Science=None),
+            ]
 
             self.assertEqual(result.collect(), expected_lst)
         except Exception:
             self.fail("Spark Connect pivot error")
 
 
+class UDFGenerationTest(SparkTestCase):
```

Comment:
@SemyonSinchenko let's move this one to tests/test_end_to_end.py like https://github.com/pyspark-ai/pyspark-ai/pull/139 did